include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -58,11 +58,11 @@ PointwiseOutputDataBase<dim>::print(dealii::ConditionalOStream & pcout) const
 template struct PointwiseOutputDataBase<2>;
 template struct PointwiseOutputDataBase<3>;
 
-template<int dim, typename VectorType>
+template<int dim, typename VectorType, typename Number>
 void
-PointwiseOutputGeneratorBase<dim, VectorType>::evaluate(VectorType const & solution,
-                                                        double const       time,
-                                                        bool const         unsteady)
+PointwiseOutputGeneratorBase<dim, VectorType, Number>::evaluate(VectorType const & solution,
```

Comment:
It looks like `PointwiseOutputGeneratorBase` does not need the `VectorType` at all, since the vector is only passed to `do_evaluate()`, but this function is purely virtual in the base class.

The circumstance that we want to be flexible regarding the VectorType and that the vectors might not always be available/accessible (see description of PR) calls for another design in  my opinion. Passing a lambda without any arguments or templates for dof-vector to the `evaluate()` function seems more natural.

The function `POGBase::evaluate()` would become protected, and concrete derived classes would implement a public function `evaluate()` for concrete vector types, which then first define the lambda and afterwards call `POGBase::evaluate(lambda)`.

Reply:
In `StatisticsManager` (https://github.com/exadg/exadg/blob/265375b649a615ffea5e4652168b801145f72c09/include/exadg/postprocessor/statistics_manager.h#L114-L121) we have defined different `evaluate_functions()` differing in the argument. Maybe such a design could be realized here?

Reply:
The suggestion with the lambda refers to this function: https://github.com/exadg/exadg/blob/9227a66efec494a3a32ccde479e7305a00849fe8/include/exadg/postprocessor/pointwise_output_generator_base.h#L125

Instead of using the vector as argument in the different functions, we could use the function itself as an argument in order to hide the type of the vector. I think we only want to be able to call an "action", while it is not that important which arguments are needed for this action. Or do I misunderstand something?